### PR TITLE
Fix test ownership test to use task_name if target is a devicelab target

### PR DESCRIPTION
**Pull request by ricardoamador**
Fix the test ownership test to compare the task_name with the dart file store in the TESTOWNERs file.

*List which issues are fixed by this PR. You must list at least one issue.*
Fixes https://github.com/flutter/flutter/issues/125328

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/lib/src/request_handlers/test_ownership.dart
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/test_ownership.dart:53]: We should use test name directly instead of parsing the test name based on target name.

Counter example:
target name: `Mac_ios spell_check_test`
task name: `spell_check_test_ios`

Comment (ricardoamador) [reply app_dart/lib/src/request_handlers/test_ownership.dart:53]: It is for the sake of the TESTOWNER validation since I am passing in just the task name from the validate test ownership file. However if this is to be applied everywhere then this will take a significant amount of time as there are multiple places we are calling this from where the task name might no be available. I will have to look again.

Comment (ricardoamador) [reply app_dart/lib/src/request_handlers/test_ownership.dart]: I received clarification in a gvc.

File: app_dart/lib/src/request_handlers/check_flaky_builders.dart
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/check_flaky_builders.dart:83]:
```suggestion
        targets.singleWhere((element) => element.name == info.name!),
```

File: app_dart/lib/src/request_handlers/file_flaky_issue_and_pr.dart
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/file_flaky_issue_and_pr.dart:77]: Ditto.

File: app_dart/lib/src/request_handlers/test_ownership.dart
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/test_ownership.dart:48]: Shall we remove this function?

File: app_dart/lib/src/request_handlers/update_existing_flaky_issues.dart
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/update_existing_flaky_issues.dart:111]: Also here.

File: app_dart/lib/src/foundation/utils.dart
Comment (keyonghan) [inline app_dart/lib/src/foundation/utils.dart:196]: Do we need this block?

Comment (ricardoamador) [reply app_dart/lib/src/foundation/utils.dart:196]: Good catch. I don't think we do. I will test.

File: app_dart/lib/src/request_handlers/test_ownership.dart
Comment (ricardoamador) [reply app_dart/lib/src/request_handlers/test_ownership.dart:48]: I do believe that we still need this for the non devicelab tests as they still use the target name to split.

**Review by keyonghan — APPROVED** [review]
LGTM.

You need to wait for the flutter PR to land first to unblock the CQ here.

**Pull request merged**
